proposals/0003-spec.md

```diff
@@ -0,0 +1,249 @@
+* The JSON MAY support an additional information holder to carry key value pairs that provide additional context
+
+
+# Health Check Procedures
+* A producer MUST support custom health check procedures
+* A producer SHOULD support reasonable out-of-the-box procedures
+* A producer without health check procedures installed MUST returns positive overall outcome (i.e. HTTP 204, no content)
+
+
+## Policies to determine the overall outcome
+When multiple procedures are installed all procedures MUST be executed and the overall outcome needs to be determined.
+* Consumers MUST support a logical conjunction policy to determine the outcome
+* Consumers MUST use the logical conjunction policy by default to determine the outcome
+* Consumers MAY support custom policies to determine the outcome
+Security
+Aspects regarding the secure access of health check information.
+
+
+* A producer MUST enforce security on all check invocations
+* A producer MAY ignore security for trusted origins (i.e. localhost)
+* HTTP Digest Auth MUST be one supported authentication mechanism
+* HTTP Digest Auth SHOULD be the default algorithm for the HTTP protocol binding
+
+# Appendix A: REST interface specifications
+
+| Context       | Verb          | Status Code  | Response |
+| ------------- |:-------------:| :-----:| -----|
+| /health       | GET           | 200, 204, 500, 503 | See Appendix B|
+
+## Status Codes:
+
+* 200 for a health check with a positive outcome
+* 204 in case no health check procedures are installed into the runtime
+* 503 in case the overall outcome is negative
+* 500 in case the consumer wasn’t able to process the health check request (i.e. error in procedure)
+
+
+# Appendix B: JSON payload specification
+
+The following table give valid health check responses:
+
+| Request | HTTP Status       | JSON Payload         | State  | Comment |
+| ------------- |:-------------:| :-----:| -----| ----|
+| /health       | 200           | Yes, see below  | UP | Check with payload |
+| /health       | 204           | No  | UP | Check without procedures installed |
+| /health       | 503           | Yes  | Down | Check failed|
+| /health       | 500           | No  | No | Request processing failed (i.e. error in procedure) |
+
+
+## JSON Schema:
+```
+{
+ "$schema": "http://json-schema.org/draft-04/schema#",
+ "type": "object",
+ "properties": {
+   "outcome": {
+     "type": "string"
+   },
+   "checks": {
+     "type": "array",
+     "items": {
+       "type": "object",
+       "properties": {
+         "id": {
+           "type": "string"
+         },
+         "result": {
+           "type": "string"
+         },
+         "data": {
+           "type": "object",
+           "properties": {
+             "key": {
+               "type": "string"
+             },
+             "value": {
+               "type": "string|boolean|int"
+             }
+           }
+         }
+       },
+       "required": [
```

Comment:
I like this proposal.  I wonder if it's good to include the time stamp in the payload to collate with the monitoring data or logs later.  Should the time stamp  be done at the consumer side instead?